plugins/samples/error_page_with_traceid/plugin.rs

```diff
@@ -0,0 +1,124 @@
+    .trace-id { 
+      background-color: #f5f5f5; 
+      padding: 1rem; 
+      font-family: monospace;
+      word-break: break-all;
+      margin-top: 2rem;
+    }
+  </style>
+</head>
+<body>
+  <div class="container">
+    <h1>Error {STATUS_CODE}</h1>
+    <p>We're sorry, something went wrong with your request.</p>
+    
+    <div class="trace-id">
+      <strong>Trace ID:</strong> {TRACE_ID}
+    </div>
+    
+    <p>Please provide this trace ID to support for assistance.</p>
+  </div>
+</body>
+</html>
+"#;
+
+proxy_wasm::main! {{
+    proxy_wasm::set_log_level(LogLevel::Debug);
+    proxy_wasm::set_http_context(|_, _| -> Box<dyn HttpContext> { Box::new(ErrorPageWithTraceId::default()) });
+}}
+
+struct ErrorPageWithTraceId {
+    trace_id: String,
+}
+
+impl Default for ErrorPageWithTraceId {
+    fn default() -> Self {
+        Self {
+            trace_id: "not-available".to_string(),
+        }
+    }
+}
+
+impl Context for ErrorPageWithTraceId {}
+
+impl HttpContext for ErrorPageWithTraceId {
+    fn on_http_request_headers(&mut self, _num_headers: usize, _end_of_stream: bool) -> Action {
+        self.trace_id = self.extract_trace_id();
+        log::debug!("Captured trace ID: {}", self.trace_id);
+        Action::Continue
+    }
+
+    fn on_http_response_headers(&mut self, _num_headers: usize, _end_of_stream: bool) -> Action {
+        if let Some(status) = self.get_http_response_header(":status") {
+            if let Ok(status_code) = status.parse::<u16>() {
+                // Only handle error status codes (4xx and 5xx)
+                if status_code >= 400 {
+                    let error_page = ERROR_TEMPLATE
+                        .replace("{STATUS_CODE}", &status)
+                        .replace("{TRACE_ID}", &self.trace_id);
+
+                    // Convert u16 to u32 for status code
+                    self.send_http_response(
+                        status_code as u32,
+                        vec![("Content-Type", "text/html; charset=utf-8")],
+                        Some(error_page.as_bytes()),
+                    );
+                    return Action::Pause;
+                }
+            }
+        }
+        Action::Continue
+    }
+}
+
+impl ErrorPageWithTraceId {
+    fn extract_trace_id(&self) -> String {
+        // Try standard Google Cloud trace header first
+        if let Some(trace_header) = self.get_http_request_header("x-cloud-trace-context") {
+            // Format: TRACE_ID/SPAN_ID;o=TRACE_TRUE
+            if let Some(slash_pos) = trace_header.find('/') {
+                return trace_header[..slash_pos].to_string();
+            }
+            return trace_header;
+        }
+
+        // Try W3C Trace Context standard
+        if let Some(w3c_trace) = self.get_http_request_header("traceparent") {
+            // Format: version-trace_id-parent_id-flags
+            let parts: Vec<&str> = w3c_trace.split('-').collect();
+            if parts.len() >= 4 && parts[1].len() == 32 {
```

Comment:
Should `>=` be `==` ? Also consider using a regex to match and extract the field, so that it can verify all chars are hex digits.